src/libs/actions/Session/index.js

```diff
@@ -422,23 +421,88 @@ function changePasswordAndSignIn(authToken, password) {
             }
             if (responsePassword.jsonCode === CONST.JSON_CODE.NOT_AUTHENTICATED) {
                 // authToken has expired, and we have the account email, so we request a new magic link.
-                Onyx.merge(ONYXKEYS.ACCOUNT, {error: null});
+                Onyx.merge(ONYXKEYS.ACCOUNT, {errors: null});
                 resetPassword();
                 Navigation.navigate(ROUTES.HOME);
                 return;
             }
-            Onyx.merge(ONYXKEYS.SESSION, {error: 'setPasswordPage.passwordNotSet'});
+            Onyx.merge(ONYXKEYS.SESSION, {errors: {[DateUtils.getMicroseconds()]: Localize.translateLocal('setPasswordPage.passwordNotSet')}});
         });
 }
 
+/**
+ * Validates new user login, sets a new password and authenticates them
+ * @param {Number} accountID
+ * @param {String} validateCode
+ * @param {String} password
+ */
+function setPasswordForNewAccountAndSignin(accountID, validateCode, password) {
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: true,
+            },
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.SESSION,
+            value: {
+                errors: null,
+            },
+        },
+    ];
+
+    const successData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: false,
+                errors: null,
+            },
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.SESSION,
+            value: {
+                errors: null,
+            },
+        },
+    ];
+
+    const failureData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: false,
```

Comment:
Does this need an `errors:` in the value here as well?

Reply:
At client side, we check for errors in both objects so I think it is fine for now
https://github.com/Expensify/App/blob/46b975ef73ede959fb707e7dfb4f63d8b2dcbb57/src/pages/SetPasswordPage.js#L103-L104

But yes, we need to move all errors to one Onyx key on both server and client side.